apps/prairielearn/src/pages/instructorAssessmentInstances/instructorAssessmentInstances.js

```diff
@@ -1,191 +1,204 @@
+      assessmentInstanceRowSchema,
+    );
+    res.send(result);
+    return;
+  }),
+);
 
-router.get('/', function (req, res, next) {
-  debug('GET /');
-  if (!res.locals.authz_data.has_course_instance_permission_view) {
-    return next(error.make(403, 'Access denied (must be a student data viewer)'));
-  }
-  debug('render page');
-  res.render(__filename.replace(/\.js$/, '.ejs'), res.locals);
-});
+router.get(
+  '/client.js',
+  asyncHandler(async (req, res) => {
+    if (!res.locals.authz_data.has_course_instance_permission_view) {
+      throw error.make(403, 'Access denied (must be a student data viewer)');
+    }
+    res.type('text/javascript');
+    res.render(__filename.replace(/\.js$/, 'ClientJS.ejs'), res.locals);
+  }),
+);
 
-router.post('/', function (req, res, next) {
-  if (!res.locals.authz_data.has_course_instance_permission_edit) {
-    return next(error.make(403, 'Access denied (must be a student data editor)'));
-  }
+router.get(
+  '/',
+  asyncHandler(async (req, res) => {
+    if (!res.locals.authz_data.has_course_instance_permission_view) {
+      throw error.make(403, 'Access denied (must be a student data viewer)');
+    }
+    debug('render page');
+    res.render(__filename.replace(/\.js$/, '.ejs'), res.locals);
+  }),
+);
 
-  if (req.body.__action === 'close') {
-    const assessment_id = res.locals.assessment.id;
-    const assessment_instance_id = req.body.assessment_instance_id;
-    checkBelongs(assessment_instance_id, assessment_id, (err) => {
-      if (ERR(err, next)) return;
+router.post(
+  '/',
+  asyncHandler(async (req, res) => {
+    if (!res.locals.authz_data.has_course_instance_permission_edit) {
+      throw error.make(403, 'Access denied (must be a student data editor)');
+    }
 
+    if (req.body.__action === 'close') {
+      const assessment_id = res.locals.assessment.id;
+      const assessment_instance_id = req.body.assessment_instance_id;
+      checkBelongsAsync(assessment_instance_id, assessment_id);
       const requireOpen = true;
       const close = true;
       const overrideGradeRate = true;
-      gradeAssessmentInstance(
+      gradeAssessmentInstanceAsync(
         assessment_instance_id,
         res.locals.authn_user.user_id,
         requireOpen,
         close,
         overrideGradeRate,
-        function (err) {
-          if (ERR(err, next)) return;
-          res.send(JSON.stringify({}));
-        },
       );
-    });
-  } else if (req.body.__action === 'delete') {
-    const assessment_id = res.locals.assessment.id;
-    const assessment_instance_id = req.body.assessment_instance_id;
-    checkBelongs(assessment_instance_id, assessment_id, (err) => {
-      if (ERR(err, next)) return;
-
+      res.send(JSON.stringify({}));
+    } else if (req.body.__action === 'delete') {
+      const assessment_id = res.locals.assessment.id;
+      const assessment_instance_id = req.body.assessment_instance_id;
+      checkBelongsAsync(assessment_instance_id, assessment_id);
```

Comment:
This is an async function; it should be awaited.

Reply:
This is now being awaited in commit [8472ae8](https://github.com/PrairieLearn/PrairieLearn/pull/9080/commits/8472ae8a0869182f87c71e15b748cbd6064db927).